sui_types/src/committee.rs

```diff
@@ -68,4 +70,34 @@ impl Committee {
         // then (N + 2) / 3 = f + 1 + k/3 = f + 1
         (self.total_votes + 2) / 3
     }
+
+    /// Given a sequence of (AuthorityName, value) for values that are ordered, provide the
+    /// value at the particular threshold by stake. This orders all provided values and pick
+    /// the appropriate value that has under it threshold stake. You may use the function
+    /// `quorum_threshold` or `validity_threshold` to pick the f+1 or 2f+1 thresholds
+    /// respectivelly.
+    pub fn robust_value<A, V>(
+        &self,
+        items: impl Iterator<Item = (A, V)>,
+        threshold: usize,
+    ) -> (AuthorityName, V)
+    where
+        A: Borrow<AuthorityName> + Ord,
+        V: Ord,
+    {
+        debug_assert!(threshold < self.total_votes);
+
+        let vec: Vec<_> = items
+            .map(|(a, v)| (v, self.voting_rights[a.borrow()], *a.borrow()))
+            .sorted()
```

Comment:
Should we sort in descending order? What would be the difference between ascending and descending?

Reply:
It really depends on the semantics of the value and what you need from the robust value. I have added more explanation on the function doc.